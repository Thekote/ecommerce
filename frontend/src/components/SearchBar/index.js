import styled from "styled-components"

const SearchBar = (props) => (
  <Input type="text" placeholder="Pesquisar" {...props} />
)

export default SearchBar

const Input = styled.input`
  height: 40px;
  width: 30rem;
  padding: 0 20px;
  border-radius: 5px;
  border: 1px solid black;
`
