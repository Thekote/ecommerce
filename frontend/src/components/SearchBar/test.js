import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event"

import SearchBar from "."

describe("<SearchBar />", () => {
  it("should match snapshot", () => {
    const { container } = render(<SearchBar />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it("should have a value of filterProduct", () => {
    render(<SearchBar value={"procurando"} onChange={() => {}} />)

    const input = screen.getByPlaceholderText(/pesquisar/i)

    expect(input.value).toBe("procurando")
  })

  it("should call handleChangeFilterProduct function on each key pressed", () => {
    const onChange = jest.fn()
    render(<SearchBar onChange={onChange} />)

    const input = screen.getByPlaceholderText(/pesquisar/i)
    const value = "o valor"

    userEvent.type(input, value)
    expect(input.value).toBe(value)
    expect(onChange).toHaveBeenCalledTimes(value.length)
  })
})
