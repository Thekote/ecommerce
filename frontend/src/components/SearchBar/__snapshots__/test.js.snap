// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> should match snapshot 1`] = `
.c0 {
  height: 40px;
  width: 30rem;
  padding: 0 20px;
  border-radius: 5px;
  border: 1px solid black;
}

<input
  class="c0"
  placeholder="Pesquisar"
  type="text"
/>
`;
